tokio-trace/tokio-trace-core/src/span.rs

```diff
@@ -130,17 +138,29 @@ impl<'a> Attributes<'a> {
         }
     }
 
-    /// Returns the new span's explicitly-specified parent, if there is one.
+    /// Returns a reference to the new span's explicitly-specified parent, if
+    /// there is one.
     ///
     /// Otherwise (if the new span is a root or is a child of the current span),
-    /// returns false.
+    /// returns none.
     pub fn parent(&self) -> Option<&Id> {
         match self.parent {
             Parent::Explicit(ref p) => Some(p),
             _ => None,
         }
     }
 
+    /// Returns the new span's explicitly-specified parent, if there is one.
+    ///
+    /// Otherwise (if the new span is a root or is a child of the current span),
+    /// returns false.
+    pub fn take_parent(self) -> Option<Id> {
```

Comment:
Why is this function added? I do not see the relation to the PR goal.

Reply:
`Attributes` may hold a parent span's  `Id`. Getting that `Id` in order to construct an `Attributes` will increment the reference count for that `Id`. The `Attributes` currently exposes a function to  borrow a reference to the parent `Id`.

When a new span is constructed, an `Attributes` is _moved_ into `Subscriber::new_span`. If the subscriber wishes to store the parent `Id` of the newly constructed span, it may call `Attributes::parent`, getting a borrowed `Id`, and then call `self.clone_span` to clone the `Id`. Since the `Attributes` was moved into `new_span`, it will be dropped when `new_span` returns, calling `drop_span` with that `Id`. However, since the `Attributes` was passed by value, the subscriber _could_ just take the parent `Id` out of the `Attributes`, avoiding a `clone_span`/`drop_span`. That was the intention behind this function.

This isn't _strictly_ necessary for this change, but since `Subscriber`s can no longer `clone` the `Id` in an `Attributes`, it would be good to provide this function for convenience. If you prefer, I can remove this and open a separate PR for it.